File: test_libm/test_libm.f90
program test_libm
implicit none
complex(8), allocatable :: a(:)
integer n
real(8), external :: omp_get_wtime
real(8) :: t
complex(8) :: z
integer j
!
n = 10000000
allocate(a(n))
!
t = -omp_get_wtime()
do j = 1, 10
  call run_test(a, n, z)
enddo
t = t + omp_get_wtime()
!
write(*,*)'time=',t,' sec.'
write(*,*)'z=',z

deallocate(a)

end program

subroutine run_test(a, n, z)
implicit none
integer, intent(in) :: n
complex(8), intent(inout) :: a(n)
complex(8), intent(out) :: z
integer :: i
real(8) d,p

z = dcmplx(0.d0, 0.d0)
d = 1.d0 / n
do i = 1, n
  p = d * i * 2 * 3.141592d0
  a(i) = exp(dcmplx(0.d0, p))
enddo
do i = 1, n
  z = z + a(i)
enddo


end subroutine

File: test_libm/test_libm.cpp
#include <complex>
#include <iostream>
#include <omp.h>

void run_test(int n, std::complex<double>& z)
{
    double d = 1.0 / n;
    for (int j = 0; j < n; j++) {
        double p = d * j * 2 * 3.141592;
        z += std::exp(std::complex<double>(0.0, p));
    }
}

int main(int argn, char** argv)
{
    int n = 10000000;
    std::complex<double> z(0, 0);

    double t = -omp_get_wtime();
    for (int j = 0; j < 10; j++) {
        run_test(n, z);
    }
    t += omp_get_wtime();
    std::cout << "time= " << t << " sec." << std::endl
              << "z=" << z << std::endl;
    return 0;
}
